android/src/main/kotlin/com/airship/flutter/ConfigManager.kt

```diff
@@ -26,42 +28,174 @@ class ConfigManager(private val context: Context) {
+                //.applyDefaultProperties(context)
+                .setRequireInitialRemoteConfigEnabled(true)
                 .apply {
-                    if (!(appCredentialsOverride.first.isNullOrEmpty() || appCredentialsOverride.second.isNullOrEmpty())) {
-                        this.setAppKey(appCredentialsOverride.first)
-                            .setAppSecret(appCredentialsOverride.second)
+                    pluginConfig.inProduction.let {
+                        println(it)
+                        this.setInProduction(it)
+                    }
+
+                    pluginConfig.development.let {
+                        this.setDevelopmentAppKey(it.appKey)
+                            .setDevelopmentAppSecret(it.appSecret)
+                            .setDevelopmentLogLevel(it.logLevel.value)
+                            .build()
+                    }
+
+                    pluginConfig.production.let {
+                        this.setProductionAppKey(it.appKey)
+                            .setProductionAppSecret(it.appSecret)
+                            .setProductionLogLevel(it.logLevel.value)
+                            .build()
+                    }
+
+                    /// since proto buf defaults to empty string,
+                    // check for empty string instead of null
+                    pluginConfig.defaultEnv.let {
+                        this.setAppKey(it.appKey)
+                            .setAppSecret(it.appSecret)
+                            .setLogLevel(it.logLevel.value)
                             .build()
                     }
+
+
+                    this.setSite(
+                        when (pluginConfig.site) {
+                            Config.Site.SITE_EU -> {
+                                AirshipConfigOptions.SITE_EU
+                            }
+                            Config.Site.SITE_US -> {
+                                AirshipConfigOptions.SITE_US
+                            }
+                            Config.Site.UNRECOGNIZED -> {
+                                AirshipConfigOptions.SITE_US
+                            }
+                            else-> AirshipConfigOptions.SITE_US
+                        }
+                    )
+
+
+
+                    pluginConfig.urlAllowListList.let {
+                        this.setUrlAllowList(null/*it.toTypedArray()*/)
+                    }
+
+                    pluginConfig.urlAllowlistScopeJavascriptInterfaceList.let {
+                        this.setUrlAllowListScopeJavaScriptInterface(it.toTypedArray())
+                    }
+
+                    pluginConfig.urlAllowListScopeOpenUrlList.let {
+
+                        this.setUrlAllowListScopeOpenUrl(it.toTypedArray())
+                    }
+
+                    this.setEnabledFeatures(pluginConfig.featuresEnabledList.supportValue())
+
+                    this.setAppStoreUri(Uri.parse(pluginConfig.android.appStoreUri))
```

Comment:
It'd be good to wrap the `Uri.parse` in a try/catch here, so we can log a helpful error message. It might be worth re-throwing so that it's obvious that something is broken, though.

Reply:
The only thing `Uri.parse` throws is a `NullPointerException` – if uriString is null 
I think just checking that would be better.